connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/ProcessingContext.java

```diff
@@ -139,6 +145,18 @@ public void report() {
         }
     }
 
+    public Future<Void> report(Callback callback) {
+        List<Future<RecordMetadata>> futures = new ArrayList<>();
+        for (ErrorReporter reporter: reporters) {
+            Future<RecordMetadata> future = reporter.report(this, callback);
+            if (!future.isDone()) {
+                futures.add(future);
+            }
+        }
+        return new ErrantRecordFuture(futures);
```

Comment:
Since we often have just one reporter, it is probably worth avoiding the unnecessary allocations:
```suggestion
        if (reporters.size() == 1) {
            return reporters.get(0).report(this);
        }
        List<Future<RecordMetadata>> futures = new LinkedList<>();
        for (ErrorReporter reporter: reporters) {
            Future<RecordMetadata> future = reporter.report(this, callback);
            if (!future.isDone()) {
                futures.add(future);
            }
        }
        if (futures.isEmpty()) {
            return CompletableFuture.completedFuture(null);
        }
        return new ErrantRecordFuture(futures);
```
And since we don't know how many futures we'll add to the list (and it will likely be just zero if the DLQ is not configured or just one for the DLQ), let's use a `LinkedList` instead to avoid excessive allocation when adding the first element to the `ArrayList`.